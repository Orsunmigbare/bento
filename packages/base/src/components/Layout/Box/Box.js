import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  targetSize,
  spacing,
  borderRadius,
  fontFamily,
  font,
  color,
  themify
} from "@staccx/theme"
const Box = ({ children, className, size, ...otherProps }) => (
  <Boxing className={className} size={size} {...otherProps}>
    {children}
  </Boxing>
)

const Boxing = styled.div`
  padding: ${p => spacing(p.size)};
`

Box.defaultProps = {
  className: "",
  size: "medium"
}

Box.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.string,
    PropTypes.array
  ]),
  className: PropTypes.string,
  size: PropTypes.oneOf([
    "huge",
    "large",
    "mediumPlus",
    "medium",
    "small",
    "tiny",
    "micro"
  ])
}

export default Box
