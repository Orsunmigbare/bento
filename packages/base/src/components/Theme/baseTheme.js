export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
