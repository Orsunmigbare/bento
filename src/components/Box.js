import styled from "styled-components"
import { spacing } from "@staccx/theme"

// todo: Make box component in bento
export const Box = styled.div`
  background: #ffffff;
  box-shadow: 0px 16px 16px rgba(0, 0, 75, 0.002),
    0px 8px 8px rgba(0, 0, 75, 0.001), 0px 4px 7px rgba(0, 0, 75, 0.021),
    0px 2px 2px rgba(0, 0, 75, 0.021), 0px 32px 22px rgba(0, 0, 75, 0.021),
    0px 44px 64px rgba(0, 0, 75, 0.0101);
  border-radius: 3px;
  padding: ${spacing.medium};
`
