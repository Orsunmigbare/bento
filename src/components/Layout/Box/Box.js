import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  applyVariants,
  spacing,
  commonPropTypes,
  themePropTypes
} from "../../../theming"

const Box = ({ children, className, size, ...otherProps }) => (
  <Boxing className={className} size={size} {...otherProps}>
    {children}
  </Boxing>
)

Box.themeProps = {
  box: {
    name: "BOX",
    description: "Box style",
    type: themePropTypes.style
  }
}

export const Boxing = styled.div`
  ${p => p.size && `padding: ${spacing(p.size)(p)}`};
  ${applyVariants(Box.themeProps.box)};
`

Box.defaultProps = {
  className: "",
  size: ""
}

Box.propTypes = {
  /**
   * The component can be nested.
   */
  children: commonPropTypes.children,
  /**
   * The class that can be assigned to the Box component.
   */
  className: PropTypes.string,
  /**
   * The size of the box.
   */
  size: commonPropTypes.spacing
}

export default Box
