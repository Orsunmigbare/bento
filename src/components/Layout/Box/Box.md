```jsx
<div>
  <Box size="small">
    <div style={{ backgroundColor: "#FFE74C", padding: 24 }}>
      Content inside box small
    </div>
  </Box>

  <Box>
    <div style={{ backgroundColor: "#FF5964", padding: 24 }}>
      Content inside box medium (default)
    </div>
  </Box>

  <Box size="large">
    <div style={{ backgroundColor: "#38618C", padding: 24, color: "white" }}>
      Content inside box large
    </div>
  </Box>

  <Box size="huge">
    <div style={{ backgroundColor: "#35A7FF", padding: 24, color: "white" }}>
      Content inside box huge
    </div>
  </Box>
</div>
```
