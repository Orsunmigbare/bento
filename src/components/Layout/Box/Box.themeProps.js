import { themePropTypes } from "../../../theming"

const themeProps = {
  box: {
    name: "BOX",
    description: "Box style",
    type: themePropTypes.style
  }
}

export default themeProps
