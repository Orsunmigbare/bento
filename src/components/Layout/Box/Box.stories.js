import React from "react"
import { storiesOf } from "@storybook/react"
import { withKnobs, select, text } from "@storybook/addon-knobs"
import Box from "./Box"
import docs from "./Box.md"

storiesOf("components|base/Layout/Box", module)
  .addDecorator(withKnobs)
  .add("No text", () => <Box />)
  .add("Long Text", () => (
    <Box>
      {" "}
      {text(
        "Text",
        "Hello Hello Hello Hello Hello Hello Hello Hello Hello Hello Hello Hello Hello Hello Hello Hello "
      )}
    </Box>
  ))
  .add("Short Text", () => <Box> {text("Text", "Hello")}</Box>)
  .add(
    "Size medium",
    () => {
      const label = "Størrelse"
      const options = {
        huge: "huge",
        large: "large",
        mediumPlus: "mediumPlus",
        medium: "medium",
        small: "small",
        tiny: "tiny",
        micro: "micro",
        flush: "flush"
      }
      const defaultValue = "medium"
      const value = select(label, options, defaultValue)
      return <Box size={value}>{text("Tekst", "Box")}</Box>
    },
    {
      info: {
        text: docs
      }
    }
  )
  .add(
    "Size huge",
    () => {
      const label = "Størrelse"
      const options = {
        huge: "huge",
        large: "large",
        mediumPlus: "mediumPlus",
        medium: "medium",
        small: "small",
        tiny: "tiny",
        micro: "micro",
        flush: "flush"
      }
      const defaultValue = "huge"
      const value = select(label, options, defaultValue)
      return <Box size={value}>{text("Tekst", "Box")}</Box>
    },
    {
      info: {
        text: docs
      }
    }
  )
  .add(
    "Size tiny",
    () => {
      const label = "Størrelse"
      const options = {
        huge: "huge",
        large: "large",
        mediumPlus: "mediumPlus",
        medium: "medium",
        small: "small",
        tiny: "tiny",
        micro: "micro",
        flush: "flush"
      }
      const defaultValue = "tiny"
      const value = select(label, options, defaultValue)
      return <Box size={value}>{text("Tekst", "Box")}</Box>
    },
    {
      info: {
        text: docs
      }
    }
  )
