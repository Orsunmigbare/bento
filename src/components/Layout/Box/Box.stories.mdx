import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks"
import Box from "./Box"

<Meta title="Components|Layout/Box" component={Box} />

# Box
This is the Box component. The component can take size as a prop to scale!

# import

```javascript
import { Box } from "@staccx/bento"
```

# Types of Boxes
There are several ways to display the Box component. The box can be sized with these props: flush, macro, tiny, small, medium, mediumPlus, large and huge.

## Nested Boxes
The Box component can be also be nested.

<Preview>
  <Story name="Nested box">
    <Box size="flush">
        <div style={{ backgroundColor: "#38618C", padding: 24, color: "white" }}>
        Content inside box flush
            <Box size="macro">
                <div style={{ backgroundColor: "#35A7FF", padding: 24, color: "white" }}>
                    Content inside box flush
                </div>
            </Box>
        </div>
    </Box>
  </Story>
</Preview>

## Flush Box

<Preview>
  <Story name="Flush">
    <Box size="flush">
        <div style={{ backgroundColor: "#38618C", padding: 24, color: "white" }}>
        Content inside box flush
        </div>
    </Box>
  </Story>
</Preview>

## Macro Box

<Preview>
  <Story name="Macro">
    <Box size="macro">
        <div style={{ backgroundColor: "#38618C", padding: 24, color: "white" }}>
        Content inside box macro
        </div>
    </Box>
  </Story>
</Preview>


## Tiny Box

<Preview>
  <Story name="Tiny">
    <Box size="tiny">
        <div style={{ backgroundColor: "#35A7FF", padding: 24 }}>
        Content inside box tiny
        </div>
    </Box>
  </Story>
</Preview>


## Small Box

<Preview>
  <Story name="Small">
    <Box size="small">
        <div style={{ backgroundColor: "#FFE74C", padding: 24 }}>
        Content inside box small
        </div>
    </Box>
  </Story>
</Preview>

## Medium Box

<Preview>
  <Story name="Medium">
    <Box size="medium">
        <div style={{ backgroundColor: "#38618C", padding: 24, color: "white" }}>
        Content inside box medium
        </div>
    </Box>
  </Story>
</Preview>

## MediumPlus Box

<Preview>
  <Story name="Medium plus">
    <Box size="mediumPlus">
        <div style={{ backgroundColor: "#FFE74C", padding: 24 }}>
        Content inside box medium plus
        </div>
    </Box>
  </Story>
</Preview>

## Large Box

<Preview>
<Story name="Large">
    <Box size="large">
        <div style={{ backgroundColor: "#38618C", padding: 24, color: "white" }}>
        Content inside box large
        </div>
    </Box>
  </Story>
</Preview>

## Huge Box

<Preview>
    <Story name="Huge">
        <Box size="huge">
            <div style={{ backgroundColor: "#35A7FF", padding: 24, color: "white" }}>
            Content inside box huge
            </div>
        </Box>
    </Story>
</Preview>

# Props for Box

<Props of={Box} />
