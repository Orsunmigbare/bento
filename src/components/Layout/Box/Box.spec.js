import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Box from "./Box"

const Child = () => <p>Body</p>

describe("Box", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Box>
              <Child />
            </Box>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("invalid Size", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Box size={100}>
              <Child />
            </Box>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Size", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Box size="huge">
              <Child />
            </Box>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
