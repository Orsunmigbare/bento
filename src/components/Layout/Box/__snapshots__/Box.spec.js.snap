// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Snapshots No props 1`] = `
<div
  className="sc-AykKC pywwI"
  size=""
>
  <p>
    Body
  </p>
</div>
`;

exports[`Box Snapshots Size 1`] = `
<div
  className="sc-AykKC dvROKY"
  size="huge"
>
  <p>
    Body
  </p>
</div>
`;

exports[`Box Snapshots invalid Size 1`] = `
<div
  className="sc-AykKC YDVGw"
  size={100}
>
  <p>
    Body
  </p>
</div>
`;
